// react-native-device-detection/index.js
import React, {
  PixelRatio,
  Platform,
  Dimensions
} from 'react-native';

const windowSize = Dimensions.get('window');

const iSizes = [
  [375, 812], //iPhone X	2017
  [375, 812], //iPhone XS	2018
  [414, 896], //iPhone XS Max	2018
  [414, 896], //iPhone XR	2018
  [414, 896], //iPhone 11	2019
  [375, 812], //iPhone 11 Pro	2019
  [414, 896], //iPhone 11 Pro Max	2019
  [360, 780], //iPhone 12 mini	2020
  [390, 844], //iPhone 12	2020
  [390, 844], //iPhone 12 Pro	2020
  [428, 926], //iPhone 12 Pro Max	2020
  [375, 812], //iPhone 13 mini	2021
  [390, 844], //iPhone 13	2021
  [390, 844], //iPhone 13 Pro	2021
  [428, 926], //iPhone 13 Pro Max	2021
  [390, 844], //iPhone 14  2022
  [428, 926], //iPhone 14 Plus 2022
  [393, 852], //iPhone 14 Pro 2022
  [430, 932], //iPhone 14 Pro Max 2022
];

class DetectDeviceService {
  constructor() {
    this.pixelDensity = PixelRatio.get();
    this.width = windowSize.width;
    this.height = windowSize.height;
    this.adjustedWidth = this.width * this.pixelDensity;
    this.adjustedHeight = this.height * this.pixelDensity;
    
    this.isPhoneOrTablet();
    this.isIosOrAndroid();
    this.detectIphoneX();
  }
  
  isPhoneOrTablet() {
    if(this.pixelDensity < 2 && (this.adjustedWidth >= 1000 || this.adjustedHeight >= 1000)) {
      this.isTablet = true;
      this.isPhone = false;
    } else if(this.pixelDensity === 2 && (this.adjustedWidth >= 1824 || this.adjustedHeight >= 1824)) {
      this.isTablet = true;
      this.isPhone = false;
    } else {
      this.isTablet = false;
      this.isPhone = true;
    }
  }
  
  isIosOrAndroid() {
    if(Platform.OS === 'ios') {
      this.isIos = true;
      this.isAndroid = false;
    } else {
      this.isIos = false;
      this.isAndroid = true;
    }
  }

  detectIphoneX(){
	 if( Platform.OS === 'ios' &&
      !Platform.isPad &&
      !Platform.isTVOS &&
      iSizes.some(([x, y]) => x == windowSize.height && y == windowSize.width || y == windowSize.height && x == windowSize.width)
      ) {
	 	this.isIphoneX = true;
	 } else {
	 	this.isIphoneX = false;
	 }

  }
}

module.exports = new DetectDeviceService();
